src/main/java/alexiil/mc/mod/pipes/blocks/BlockPipe.java

```diff
@@ -62,8 +62,8 @@
             FACE_CENTER_SHAPES[dir.ordinal()] = VoxelShapes.union(faceShape, CENTER_SHAPE);
         }
 
-        SHAPES = new VoxelShape[2 * 2 * 2 * 2 * 2 * 2];
-        for (int c = 0; c < 0b111_111; c++) {
+        SHAPES = new VoxelShape[2 * 2 * 2 * 2 * 2 * 2 + 1];
+        for (int c = 0; c <= 0b111_111; c++) {
```

Comment:
I think it should really be `c <= 0b111_111` as the array is already big enough (as it's probably returning null from the last entry in the array, rather than throwing an index out of bounds exception).

Reply:
Yeah I see, I was better safe than sorry while trying this out though. I could change this once I get home this evening.

Reply:
Ok! Other than that this looks good, thanks!